File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        
        Map<String, Integer> map= new HashMap<>();
        
        for(int i=0; i<arr.length; i++)
        {
            if(map.containsKey(arr[i]))
            {
                map.put(arr[i], map.get(arr[i])+1);
            }
            else
            {
                map.put(arr[i],1);
            }
        }
        int count=0;
        for(int i=0; i<arr.length; i++)
        {
            if(map.get(arr[i])==1 && ++count==k)
            {
                return arr[i];
            }
        }
        return "";
    }
}

File: 2053-kth-distinct-string-in-an-array/2053-kth-distinct-string-in-an-array.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
      unordered_map<string, int> map;
        int count=0;
        
        for(int i=0; i<arr.size(); i++)
        {
            map[arr[i]]++;
        }
        
        for(int i=0; i<arr.size(); i++)
        {
            if(map[arr[i]]==1 && ++count==k)
            {
                return arr[i];
            }
        }
        return "";
    }
};